### DISCUSSION TITLE
Binding to 0.0.0.0 can not be reached by other machine in the same network

### DISCUSSION THREAD

**Discussion by sandangel** [tokio-rs/axum#2428] [Q&A]
- [x] I have looked for existing issues (including closed) about this

### Version

0.6.20

### Platform

Darwin Aarch64

### Description

I have 2 machine in the same network with IPs:

    172.16.129.1 <-- running Axum
    172.16.129.136

When running with socket bind to 0.0.0.0, it can not be reached by other machines in the same network. Within the same machine, I can check curl 172.17.129.1:3000. But from the machine 172.17.129.136, I can not run curl 172.17.129.1:3000

Original issue is here: https://github.com/huggingface/text-embeddings-inference/issues/110

But I found they just use axum, so I think I should report the issue here.

Since I'm using a mac, a high chance is that the mac is blocking access from other machine as well. I'm not sure how to check that.

**Comment by vi** [comment]
Why do you think the issue is Axum-specific? Do other web servers work properly?

Maybe it is some firewall? Have you checked what happens at a packet level?
